### PR TITLE
Resolves #171: Added automake option foreign to allow for github docu…

### LINKED ISSUE #171: autogen README error
It was naive of me to remove the README symlink without a fresh build. Automake relies on old standards to define the document structure in the file, one of these being README (no extension). This is common and resolved by setting `AUTOMAKE_OPTIONS = foreign`, which will allow us to take advantage of the GitHub standard document structure.

### PR DESCRIPTION
**Pull request by idinwoodie**
…mentation standards

Signed-off-by: Ian Dinwoodie <dinwoodieian@gmail.com>

### CODE REVIEW

**Comment by idinwoodie** [comment]
@brasko I verified a clean build with this change. I am currently working on an automated build for cgdb so that we can easily verify patches do not break builds, hence the email I sent this evening.

**Comment by brasko** [comment]
I've got to read more about what this option does before committing

**Comment by brasko** [comment]
My guess is it's better to have the README symlink than the foreign keyword. 
My opinion is based mostly on the fact that foreign is so poorly documented.

https://www.gnu.org/software/automake/manual/automake.html

Want to portability warnings above mean?

**Comment by idinwoodie** [comment]
@brasko Ignore the jumbled label mess above, I did't realize it was going to save all of the label removals as I played around with their color and text.

I agree with that we should look further into the implications of the `foreign` option so that we aren't continually chasing a cascade of problems that stemmed simply from the removal of a symlink.

From [section 3.2 Strictness](https://www.gnu.org/software/automake/manual/automake.html#Strictness):

> `foreign` - Automake will check for only those things that are absolutely required for proper operations. For instance, whereas GNU standards dictate the existence of a NEWS file, it will not be required in this mode. **This strictness will also turn off some warnings by default (among them, portability warnings).** The name comes from the fact that Automake is intended to be used for GNU programs; these relaxed rules are not the standard mode of operation.

From [section 17.1 Options generalities](https://www.gnu.org/software/automake/manual/automake.html#Options-generalities):

> As a general rule, strictness-implied warnings are overridden by those specified by explicit options. For example, even if ‘portability’ warnings are disabled by default in foreign strictness, an usage like this will end up enabling them: `AUTOMAKE_OPTIONS = -Wportability foreign`

From the [automake man page](https://linux.die.net/man/1/automake):

> Warning categories include:
**'gnu' - GNU coding standards (default in gnu and gnits modes)**
'obsolete' - obsolete features or constructions
'override' - user redefinitions of Automake rules or variables
**'portability' - portability issues (default in gnu and gnits modes)**
'syntax' - dubious syntactic constructs (default)
'unsupported' - unsupported or incomplete features (default)
'all' - all the warnings
'no-CATEGORY' - turn off warnings in CATEGORY
'none' - turn off all the warnings
'error' - treat warnings as errors

So the only warning categories we are disabling with the `foreign` mode are `gnu` and `portability`. We can reenable the portability warnings as mentioned in section 17.1 above. The the last point of concern would be what exactly the `gnu` category we are disabling entails.

From [section 22 The effect of --gnu and --gnits](https://www.gnu.org/software/automake/manual/automake.html#Gnits)

> The --gnu option (or gnu in the AUTOMAKE_OPTIONS variable) causes automake to check the following:
>- The files INSTALL, NEWS, README, AUTHORS, and ChangeLog, plus one of COPYING.LIB, COPYING.LESSER or COPYING, are required at the topmost directory of the package.
>- The options no-installman and no-installinfo are prohibited.

**In conclusion** I think it would be safe to implement `AUTOMAKE_OPTIONS = -Wportability foreign` so that we can circumvent the rigid document structure in favor of GitHub's document structure while retaining all non-document related warnings. What are your thoughts on the matter?

**Comment by idinwoodie** [comment]
@brasko enabled portability warnings if you find the explanation above acceptable

**Comment by idinwoodie** [comment]
Approved by @brasko and ready for integration.
